Scenarios/CGCT/CGCT/DemoScenarioTextToPath.m

```diff
@@ -112,35 +155,39 @@ static void drawZoomText(CGContextRef context,
         }
     }
 
-    // Draw the polygons
+    DrawnPoly* poly = [[DrawnPoly alloc] init];
+    poly->_polyPaths = [[NSMutableArray alloc] init];
+    // Draw the polygons to a path
     for (int k = 0; k < [threeDPoints count]; k++) {
+        struct PolyPathWithColors finalPolyPath;
         struct ThreeDPoly polyToDraw;
         NSValue* valPoly = [threeDPoints objectAtIndex:k];
         [valPoly getValue:&polyToDraw];
 
-        CGContextMoveToPoint(context, (polyToDraw.point1.x + toPoint.x) / 2.0, (polyToDraw.point1.y + toPoint.y) / 2.0);
-        CGContextAddLineToPoint(context, (polyToDraw.point2.x + toPoint.x) / 2.0, (polyToDraw.point2.y + toPoint.y) / 2.0);
-        CGContextAddLineToPoint(context, polyToDraw.point2.x, polyToDraw.point2.y);
-        CGContextAddLineToPoint(context, polyToDraw.point1.x, polyToDraw.point1.y);
-        CGContextClosePath(context);
+        CGMutablePathRef polyPath = CGPathCreateMutable();
+        CGPathMoveToPoint(polyPath, NULL, (polyToDraw.point1.x + toPoint.x) / 2.0, (polyToDraw.point1.y + toPoint.y) / 2.0);
+        CGPathAddLineToPoint(polyPath, NULL, (polyToDraw.point2.x + toPoint.x) / 2.0, (polyToDraw.point2.y + toPoint.y) / 2.0);
+        CGPathAddLineToPoint(polyPath, NULL, polyToDraw.point2.x, polyToDraw.point2.y);
+        CGPathAddLineToPoint(polyPath, NULL, polyToDraw.point1.x, polyToDraw.point1.y);
+        CGPathCloseSubpath(polyPath);
+
+        finalPolyPath.path = polyPath;
+        finalPolyPath.r = r * polyToDraw.grayScale;
+        finalPolyPath.g = g * polyToDraw.grayScale;
+        finalPolyPath.b = b * polyToDraw.grayScale;
 
-        CGContextSetRGBFillColor(context, polyToDraw.grayScale * r, polyToDraw.grayScale * g, polyToDraw.grayScale * b, 1);
-        CGContextDrawPath(context, zoomMode);
+        [poly->_polyPaths addObject:[NSValue valueWithBytes:&finalPolyPath objCType:@encode(struct PolyPathWithColors)]];
```

Comment:
clever, but dangerous. the CGPaths here are not adequately memory-managed since they're captured as raw byte values